src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -4034,6 +4035,8 @@ public Response addGlobusFilesToDataset(@Context ContainerRequestContext crc,
             return wr.getResponse();
         }
         
+        // @todo check if the dataset is already locked!
```

Comment:
You meant "of a given type", right? The addDatasetLock() already has a check for an existing lock of the same type (then it will just return the existing lock instead of creating a new one). 
As for implementing this as a hard constraint - are we positive we'll never want multiple locks of the same type - some workflows maybe? 
I addressed this specific situation with a separate lock check. We don't check for locks consistently in our APIs, I'm assuming under the assumption that it will be done when the the relevant commands are executed. But in this specific case, there is a potential for doing a huge amount of work before that UpdateDatasetVersionCommand is called in the end.